### PR TITLE
feat: Allow filtering of table select column label

### LINKED ISSUE #36635: [Bug]: Table with select column search and filter over the value, not over the label
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When searching or filtering over a table with a select column type, those filters are applied using the `value` value of the select, but they should use the `label` value, i.e., what the user sees in the table.
imagen

### Steps To Reproduce

1. Create a table widget.
2. Assign data, e.g., `{{ [{ "column": "foo", }] }}`.
3. Set column data type to `select`.
4. Set options for this table, e.g., `{{ { "label":"bar", "value":"foo" } }}`.
5. When searching or filtering `bar`, nothing appears. When searching or filtering `foo`, the row with the value `bar` appears.

### Public Sample App

_No response_

### Environment

Production

### Severity

Medium (Frustrating UX)

### Issue video log

_No response_

### Version

Appsmith v1.38

### PR DESCRIPTION
**Pull request by jacquesikot**
## Description

**Problem**
When filtering a table with a select column type, users expect to filter by the visible label values shown in each cell. Currently, however, filtering is applied to the underlying option values rather than the displayed labels, leading to unexpected filter results for end-users.

**Root Cause**
In a previous update ([PR #35124](https://github.com/appsmithorg/appsmith/pull/35124)), the table cell display for select columns was changed to show labels instead of values. However, the filtering logic was not updated accordingly, so the table still filtered on the original option values, creating a mismatch between displayed and filtered content.

**Solution**
This PR modifies the displayedRow property within the table widget to use the label property instead of the value key when filtering or searching select column data. This ensures that table filtering and searching now align with the visible label values in the select columns, providing a more intuitive user experience.

Fixes #36635 

## Automation

/ok-to-test tags="@tag.Sanity, @tag.Table, @tag.Select, @tag.Binding"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11234735437>
> Commit: fd6c179ffb2c61d23cb98fd749c8df49cebcfcdd
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11234735437&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity, @tag.Table, @tag.Select, @tag.Binding`
> Spec:
> <hr>Tue, 08 Oct 2024 12:48:01 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a new test case to verify filtering functionality for the "role" column in the Table Widget.
	- Enhanced filtering mechanism to support multiple label values for select columns.

- **Bug Fixes**
	- Removed outdated search functionality to streamline user experience.

- **Refactor**
	- Restructured existing test cases for improved clarity and flow.

### CODE REVIEW

File: app/client/src/widgets/TableWidgetV2/widget/derived.js [hunk 1]
```diff
@@ -684,7 +684,7 @@ export default {
        * For select columns with label and values, we need to include the label value
        * in the search
        */
-      let labelValueForSelectCell = "";
+      let labelValuesForSelectCell = {};
```
Comment (rahulbarwal) [inline app/client/src/widgets/TableWidgetV2/widget/derived.js:687]: Please update the comment on top of this line. And check if there are any other comments that can be updated.

**Review by rahulbarwal — DISMISSED** [review]
Lets update all the comments. Otherwise good to go.

File: app/client/src/widgets/TableWidgetV2/widget/derived.js (hunk 1 above)
Comment (jacquesikot) [reply app/client/src/widgets/TableWidgetV2/widget/derived.js:687]: Thanks, its done

**Review by rahulbarwal — APPROVED** [review]

**Review by sagar-qa007 — APPROVED** [review]

**Pull request merged**